src/runtime/native/runtime.ml

```diff
@@ -101,24 +124,44 @@ let dump_counters_exn =
   Common.write_runtime_data
 
 let dump () =
-  match file_channel () with
-  (* Failure "TypeError: runtime.unix_open is not a function" indicates that the
-     code was compiled by js_of_ocaml. In this case, we *want* to do nothing at
-     process exit. See https://github.com/aantron/bisect_ppx/issues/334. *)
-  | exception Failure _ -> ()
-  | None -> ()
-  | Some channel ->
-      (try
-        dump_counters_exn channel
-      with _ ->
-        verbose Unable_to_write_file);
-      close_out_noerr channel
+  if not !bisect_file_written then
+    (if !sigterm_enable then
+       ignore @@ Unix.(sigprocmask SIG_BLOCK [Sys.sigterm]) ;
+     bisect_file_written := true;
```

Comment:
i set it to true regardless whether the write succeeds or not, not sure that makes sense.

Reply:
I'm not sure I understand what you mean here. Actually, we need just need to track if `dump` has been called to avoid writing the trace twice. However, I've changed the logic a bit in the latest version, perhaps it is more clear there.